algorithms/ar-lps/java/LongestPalindromicSubstring.java

```diff
@@ -0,0 +1,89 @@
+import java.util.ArrayList;
+import java.util.Collections;
+/**
+ *  Implementation of Longest Palindromic Substring
+ *  in Linear Time Complexity using Manachar Algorithm.
+ *  @author aayushbest
+ */
+public class LongestPalindromicSubstring{
+	
+	/**
+	 * <p>Stores the even-length of each palindrome whereas
+	 * index will work as center of the palindrome.</p>
+	 */
+	private static ArrayList<Integer> evenLengthPalindromes;
+	
+	/**
+	 * <p>Stores the odd-length of each palindrome whereas
+	 * index will work as center of the palindrome.</p>
+	 */
+	private static ArrayList<Integer> oddLengthPalindromes;
+	
+	/**
+	 *  Manacher's Algorithm - Algorithm to find longest palindrome substring
+	 *  in O(n) (Linear Time) complexity.
+	 */
+	public static String longestPalindromicSubstring(final String s) {
```

Comment:
Codacy found an issue: [The method 'longestPalindromicSubstring(String)' has an NPath complexity of 867, current threshold is 200](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6417291)